core/trino-main/src/test/java/io/trino/type/TestArrayOperators.java

```diff
@@ -435,13 +436,9 @@ public void testArrayToArrayConcat()
         // This query is ambiguous. The result can be [[1], NULL] or [[1], [NULL]] depending on interpretation
         assertInvalidFunction("ARRAY [ARRAY [1]] || ARRAY [NULL]", AMBIGUOUS_FUNCTION_CALL);
 
-        try {
-            assertFunction("ARRAY [ARRAY [1]] || ARRAY [ARRAY ['x']]", new ArrayType(new ArrayType(INTEGER)), null);
-            fail("arrays must be of the same type");
-        }
-        catch (RuntimeException e) {
-            // Expected
-        }
+        assertThatThrownBy(() -> assertFunction("ARRAY [ARRAY [1]] || ARRAY [ARRAY ['x']]", new ArrayType(new ArrayType(INTEGER)), null))
+                .isInstanceOf(TrinoException.class)
+                .hasMessage("line 1:19: Unexpected parameters (array(array(integer)), array(array(varchar(1)))) for function concat. Expected: concat(char(x), char(y)) , concat(array(E), E) E, concat(E, array(E)) E, concat(array(E)) E, concat(varchar) , concat(varbinary) ");
 
```

Comment:
may be better to do with assertTrinoExceptionThrownBy and validate also error code?